pkg/adapter/nimbus-kyverno/manager/manager.go

```diff
@@ -0,0 +1,363 @@
+			return
+		}
+		if err != nil {
+			if errors.IsNotFound(err) {
+				if err = k8sClient.Create(ctx, &kcp); err != nil {
+					logger.Error(err, "failed to create KyvernoClusterPolicy", "KyvernoClusterPolicy.Name", kcp.Name)
+					return
+				}
+				logger.Info("KyvernoClusterPolicy created", "KyvernoClusterPolicy.Name", kcp.Name)
+			}
+		} else {
+			kcp.ObjectMeta.ResourceVersion = existingKcp.ObjectMeta.ResourceVersion
+			if err = k8sClient.Update(ctx, &kcp); err != nil {
+				logger.Error(err, "failed to configure existing KyvernoClusterPolicy", "KyvernoClusterPolicy.Name", existingKcp.Name)
+				return
+			}
+			logger.Info("KyvernoClusterPolicy configured", "KyvernoClusterPolicy.Name", existingKcp.Name)
+		}
+
+		if err = adapterutil.UpdateCnpStatus(ctx, k8sClient, "KyvernoClusterPolicy/"+kcp.Name, cnp.Name, false); err != nil {
+			logger.Error(err, "failed to update KyvernoClusterPolicies status in NimbusPolicy")
+		}
+	}
+}
+
+func deleteKp(ctx context.Context, npName, npNamespace string) {
+	logger := log.FromContext(ctx)
+	var kps kyvernov1.PolicyList
+
+	if err := k8sClient.List(ctx, &kps, &client.ListOptions{Namespace: npNamespace}); err != nil {
+		logger.Error(err, "failed to list KyvernoPolicies")
+		return
+	}
+
+	// Kubernetes GC automatically deletes the child when the parent/owner is
+	// deleted. So, we don't need to do anything in this case since NimbusPolicy is
+	// the owner and when it gets deleted corresponding kps will be automatically
+	// deleted.
+	for _, kp := range kps.Items {
+		logger.Info("KyvernoPolicy already deleted due to NimbusPolicy deletion",
+			"KyvernoPolicy.Name", kp.Name, "KyvernoPolicy.Namespace", kp.Namespace,
+			"NimbusPolicy.Name", npName, "NimbusPolicy.Namespace", npNamespace,
+		)
+	}
+}
+
+func deleteDanglingkps(ctx context.Context, np intentv1.NimbusPolicy, logger logr.Logger) {
+	var existingkps kyvernov1.PolicyList
+	if err := k8sClient.List(ctx, &existingkps, client.InNamespace(np.Namespace)); err != nil {
+		logger.Error(err, "failed to list KyvernoPolicies for cleanup")
+		return
+	}
+
+	var kpsOwnedByNp []kyvernov1.Policy
+	for _, kp := range existingkps.Items {
+		ownerRef := kp.OwnerReferences[0]
+		if ownerRef.Name == np.Name && ownerRef.UID == np.UID {
+			kpsOwnedByNp = append(kpsOwnedByNp, kp)
+		}
+	}
+	if len(kpsOwnedByNp) == 0 {
+		return
+	}
+
+	kpsToDelete := make(map[string]kyvernov1.Policy)
+
+	// Populate owned kps
+	for _, kpOwnedByNp := range kpsOwnedByNp {
+		kpsToDelete[kpOwnedByNp.Name] = kpOwnedByNp
+	}
+
+	for _, nimbusRule := range np.Spec.NimbusRules {
+		kpName := np.Name + "-" + strings.ToLower(nimbusRule.ID)
+		delete(kpsToDelete, kpName)
+	}
+
+	for kpName := range kpsToDelete {
+		kp := kpsToDelete[kpName]
+		if err := k8sClient.Delete(ctx, &kp); err != nil {
+			logger.Error(err, "failed to delete dangling KyvernoPolicy", "KyvernoPolicy.Name", kp.Namespace, "KyvernoPolicy.Namespace", kp.Namespace)
+			continue
+		}
+
```

Comment:
After deleting a policy we have to update the corresponding nimbuspolicy status subresource to reflect the changes. Please call the API to update it.
For ref:
https://github.com/5GSEC/nimbus/blob/b91563b0ccd36deee1c6afd73cf51fd2aa25aae8/pkg/adapter/nimbus-netpol/manager/netpols_manager.go#L213-L215